pkg/diagnose/firewall.go

```diff
@@ -118,3 +135,164 @@ func getActiveGatewayNodeName(clusterInfo *cluster.Info, hostname string, status
 
 	return ""
 }
+
+func getGatewayIP(clusterInfo *cluster.Info, localClusterID string, status reporter.Interface) string {
```

Comment:
This function should return an error and let the caller report it appropriately.